fw/http_parser.c

```diff
@@ -8221,8 +8254,10 @@ __h2_req_parse_if_msince(TfwHttpMsg *msg, unsigned char *data, size_t len,
 	 * header field is defined as a comma-separated list [i.e., #(values)]
 	 * or the header field is a well-known exception.
 	 */
-	if (unlikely(req->cond.flags & TFW_HTTP_COND_IF_MSINCE))
+	if (unlikely(req->cond.flags & TFW_HTTP_COND_IF_MSINCE)) {
+		parser->cstate.is_set = 0;
```

Comment:
> But what if this field is of type "xxx Field With Indexing" which goes into the dynamic table? From my understanding this would modify dynamic table and if cstate.is_set is set, then that header would have some weird value associated with it.

If we will "clear" `parser->cstate` when header parsed and added to dynamic table then there will be no problem. Also, each "duplicated" header is new record  in dynamic table.

Reply:
Tempesta must drop the request when meets second If-Modified-Since in same requests. It implies this must work with headers from dynamic table as well.